packages/cli/src/utils/format.ts

```diff
@@ -0,0 +1,33 @@
+import chalk from "chalk";
+import prettier from "prettier";
+import prettierPluginSolidity from "prettier-plugin-solidity";
+
+export async function formatSolidity(content: string, prettierConfigPath?: string) {
+  let config;
+  if (prettierConfigPath) {
+    config = await prettier.resolveConfig(prettierConfigPath);
+  }
+  try {
+    return prettier.format(content, {
+      plugins: [prettierPluginSolidity],
+      parser: "solidity-parse",
+
+      printWidth: 120,
+      semi: true,
+      tabWidth: 2,
+      useTabs: false,
+      bracketSpacing: true,
+
+      ...config,
+    });
+  } catch (error) {
+    let message;
+    if (error instanceof Error) {
+      message = error.message;
+    } else {
+      message = error;
+    }
+    console.log(chalk.yellow(`Error during output formatting: ${message}`));
+    return content;
+  }
+}
```

Comment:
Not blocking but we can also call out to `forge fmt` and use the project's local `foundry.toml` (in the same way that we use it for `src` etc. dirs)

```
cat File.sol | forge fmt --raw -
```

Reply:
This method isn't really file-related, it's used at the render stage, so it's not aware of the filesystem at all, and when renderer is standalone it could even be used without files at all.
Also since it's all js using prettier is just more convenient.
I think if people want custom formatting they can do their own `fmt`, this func is just for new mudders who don't have that set up yet, so they don't see super ugly files